### PR TITLE
[cdc-cli] When using miniCluster, don't check 'FLINK_HOME'.

### PR DESCRIPTION
**Pull request by joyCurry30**
If 'FLINK_HOME' is not set, we can use empty configuration, rather than throwing exception.
cc @lvyanquan

### CODE REVIEW

File: flink-cdc-cli/src/main/java/com/ververica/cdc/cli/CliFrontend.java [hunk 1]
```diff
@@ -105,25 +104,34 @@ static CliExecutor createExecutor(CommandLine commandLine) throws Exception {
                 additionalJars);
     }
 
-    private static Path getFlinkHome(CommandLine commandLine) {
+    private static Configuration loadFlinkConfiguration(CommandLine commandLine) throws Exception {
+        Optional<Path> flinkHome = getFlinkHome(commandLine);
+        if (flinkHome.isPresent()) {
+            return FlinkEnvironmentUtils.loadFlinkConfiguration(flinkHome.get());
+        }
+        return new Configuration();
```
Comment (lvyanquan) [inline flink-cdc-cli/src/main/java/com/ververica/cdc/cli/CliFrontend.java:112]: We use Configuration from Flink conf to open checkpoint and submit job to rest client, if Configuration is empty, job could not be submitted to remote cluster, see https://github.com/ververica/flink-cdc-connectors/blob/33b7c85093a69382d23c309a9b209ddae27a2968/flink-cdc-composer/src/main/java/com/ververica/cdc/composer/flink/FlinkPipelineComposer.java#L60.

File: flink-cdc-cli/src/main/java/com/ververica/cdc/cli/CliFrontend.java [hunk 2]
```diff
@@ -105,25 +104,34 @@ static CliExecutor createExecutor(CommandLine commandLine) throws Exception {
...
+    }
+
+    private static Optional<Path> getFlinkHome(CommandLine commandLine) {
         // Check command line arguments first
         String flinkHomeFromArgs = commandLine.getOptionValue(CliFrontendOptions.FLINK_HOME);
         if (flinkHomeFromArgs != null) {
             LOG.debug("Flink home is loaded by command-line argument: {}", flinkHomeFromArgs);
-            return Paths.get(flinkHomeFromArgs);
+            return Optional.of(Paths.get(flinkHomeFromArgs));
```
Comment (lvyanquan) [inline flink-cdc-cli/src/main/java/com/ververica/cdc/cli/CliFrontend.java:120]: Paths#get will never return null, it's better to use Files#exists judgement.

File: flink-cdc-cli/src/test/java/com/ververica/cdc/cli/CliFrontendTest.java [hunk 3]
```diff
@@ -83,16 +83,6 @@ void testGlobalPipelineConfigParsing() throws Exception {
...
     }
 
-    @Test
-    void testAdditionalJar() throws Exception {
-        String aJar = "/foo/jar/a.jar";
-        String bJar = "/foo/jar/b.jar";
-        CliExecutor executor =
-                createExecutor(
-                        pipelineDef(), "--flink-home", flinkHome(), "--jar", aJar, "--jar", bJar);
-        assertThat(executor.getAdditionalJars()).contains(Paths.get(aJar), Paths.get(bJar));
```
Comment (lvyanquan) [inline flink-cdc-cli/src/test/java/com/ververica/cdc/cli/CliFrontendTest.java:93]: This test is use to verify the correctness of `--jar` option, it's better to remain it.

**Comment by lvyanquan** [comment]
Overall LGTM. Can you add a test case like [CliFrontendTest#testMissingFlinkHome](https://github.com/ververica/flink-cdc-connectors/blob/08d6b8178ffe89f787067e9e0cc7324b0e66fb97/flink-cdc-cli/src/test/java/com/ververica/cdc/cli/CliFrontendTest.java#L64)?

**Review by lvyanquan — APPROVED** [review]

**Comment by lvyanquan** [comment]
@joyCurry30 To go a step further, can you add a section in [tutorial](https://ververica.github.io/flink-cdc-connectors/master/content/quickstart/mysql-doris-pipeline-tutorial.html) to introduce this `--use-mini-cluster` parameter?

File: flink-cdc-cli/src/main/java/org/apache/flink/cdc/cli/CliFrontend.java [hunk 4]
```diff
@@ -86,6 +87,18 @@ static CliExecutor createExecutor(CommandLine commandLine) throws Exception {
         // Global pipeline configuration
         Configuration globalPipelineConfig = getGlobalConfig(commandLine);
 
+        // Create pipeline composer
+        boolean useMiniCluster = commandLine.hasOption(CliFrontendOptions.USE_MINI_CLUSTER);
```
Comment (PatrickRen) [inline flink-cdc-cli/src/main/java/org/apache/flink/cdc/cli/CliFrontend.java:91]: How could the MiniCluster run without specifying `FLINK_HOME`? We don't package `flink-dist` into our distribution, so I assume that the MiniCluster cannot even be started without specifying the place where Flink related classes are.

**Review by PatrickRen — COMMENTED** [review]
@joyCurry30 Thanks for the contribution. I left a comment.

**Pull request closed without merging**